include/re_types.h

```diff
@@ -263,9 +263,13 @@ typedef bool _Bool;
 #define likely(x)       __builtin_expect(!!(x), 1)
 #define unlikely(x)     __builtin_expect(!!(x), 0)
 
-
+/* Socket helpers */
 #ifdef WIN32
 #define ERRNO_SOCKET WSAGetLastError()
+#define BAD_SOCKET INVALID_SOCKET
+#define RE_SOCKET SOCKET
 #else
 #define ERRNO_SOCKET errno
+#define BAD_SOCKET -1
+#define RE_SOCKET int
```

Comment:
is it possible to use typedef instead of define ?

we should consider placing these socket helpers in re_net.h
the helpers will mainly be used by udp.c and tcp.c

Reply:
Yes I have replaced locally by a typedef. Was a bit sick the last days, will update this PR shortly.